pdf_form_fill/pdf_form_fill.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2010-2013 OpenERP s.a. (<http://openerp.com>).
+#    Copyright (C) 2013 initOS GmbH(<http://www.initos.com>).
+#    Author Nikolina Todorova <nikolina.todorova@initos.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+try:
+    from fdfgen import forge_fdf
+except ImportError:
+    forge_fdf = False
+from openerp.osv import orm
+from tempfile import NamedTemporaryFile
+import subprocess
+import base64
+import os
+
+
+class pdf_form_fill(orm.Model):
+    _name = 'pdf.form.fill'
+
+    def get_directory_name(self):
+        return os.path.dirname(__file__)
+
+    def prepare_pdf_fields(self, cr, uid, ids, context=None):
+        """Overwrite this to add specific preparations.
+           Return value has to be a list of tupel ('pdf field name', value)
+        """
+        return []
+
+    def get_pdf_file_path(self, cr, uid, ids, context=None):
+        """Overwrite this to return the specific pdf file path."""
+        raise orm.except_orm(
+            (u'Not implemented Error'),
+            (u'There is no file path method for the given model.')
```

Comment:
at the top of this module add this line:

    from openerp.tools.translate import _

Then modify the 3 lines above as follows:

        raise orm.except_orm(
            _(u'Not implemented Error'),
            _(u'There is no file path method for the given model.')

This marks the strings as translatable.